parser/parser.y

```diff
@@ -4804,15 +4810,15 @@ Statement:
 |	UnionStmt
 |	SetStmt
 |	ShowStmt
-|	TruncateTableStmt
-|	UpdateStmt
-|	UseStmt
 |	SubSelect
 	{
 		// `(select 1)`; is a valid select statement
 		// TODO: This is used to fix issue #320. There may be a better solution.
-		$$ = $1.(*ast.SubqueryExpr).Query
+		$$ = $1.(*ast.SubqueryExpr).Query.(ast.StmtNode)
```

Comment:
why we add type assertion here?

Reply:
$1.(*ast.SubqueryExpr).Query is a ResultSet, not StmtNode.